### ISSUE TITLE
MA0048 Alternative - File name must match first type name

### ISSUE THREAD

**Issue by SebastianStehle** [meziantou/Meziantou.Analyzer#297]
The Rule "MA0048 - File name must match type name" works great, but in some cases I have additional types in my file.

e.g. I usually do not create a separate file just to define a Delegate or simple record.

Would be great to have a small variation of this rule to check only the first type.

**Comment by meziantou** [comment]
If one file contains many types, wouldn't it be easier to disable the rule in this file?

````c#
class Type1 {}

#pragma warning disable MA0048
// all other types
class Type2 {}
````

**Comment by SebastianStehle** [comment]
Sure, I could ;) 

I just realized that stylecop has this analyzer but it fails: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/4d9b3e3bb785a55f73b3029a843f0c0b73cc9ea7/documentation/SA1649.md

**Comment by CheloXL** [comment]
I have another case. I have, por example, a class ClassName:IClass and ClassName<T>:IClass. I have both classes in different files, where the first file is named like the class name, and the second file is named ClassName´1 (it's common to name the file based on the number of type arguments when they all share the same name).

Yes, I can disable the warning with #pragma, but I would prefer if the analyser can recognize this pattern (and in fact, let me fix if the file is not named accordling).

**Comment by meziantou** [comment]
@CheloXL The analyzer already recognizes this pattern except it uses the `` ` `` _separator_ instead of `´`. `` ` `` is the separator used for `MetadataName` in Roslyn. The analyzer also supports `ClassName{T}.cs`, `ClassNameOfT.cs`, `ClassName`1.whatever.cs`.

I'm not sure `´` is common. cs.github.com shows very few results and in only two different projects (my query may be wrong). That's being said, you can submit a PR if you want the analyzer to support `´`.

**Comment by CheloXL** [comment]
Ahhh, thanks. Will change my default. Not sure from where I got the `´`.
